src/generate_tangent.jl

```diff
@@ -0,0 +1,36 @@
+"""
+    rand_tangent([rng::AbstractRNG,] x)
+
+Returns a randomly generated tangent vector appropriate for the primal value `x`.
+"""
+rand_tangent(x) = rand_tangent(Random.GLOBAL_RNG, x)
+
+function rand_tangent(rng::AbstractRNG, x::Union{Symbol, AbstractChar, AbstractString})
+    return DoesNotExist()
+end
+
+rand_tangent(rng::AbstractRNG, x::Integer) = DoesNotExist()
+
+rand_tangent(rng::AbstractRNG, x::T) where {T<:Number} = randn(rng, T)
+
+rand_tangent(rng::AbstractRNG, x::StridedArray) = rand_tangent.(Ref(rng), x)
+
+function rand_tangent(rng::AbstractRNG, x::T) where {T<:Tuple}
+    return Composite{T}(rand_tangent.(Ref(rng), x)...)
+end
+
+function rand_tangent(rng::AbstractRNG, xs::T) where {T<:NamedTuple}
+    return Composite{T}(; map(x -> rand_tangent(rng, x), xs)...)
+end
+
+function rand_tangent(rng::AbstractRNG, x::T) where {T}
```

Comment:
Probably we only should be supporting struct types.
Not `primative` types.
```suggestion
function rand_tangent(rng::AbstractRNG, x::T) where {T}
    isstructtype(T) || ArgumentError("Non-struct types are not supported by this fallback.")
```